flake.nix

```diff
@@ -83,7 +83,10 @@
         };
 
         cargoArtifacts = craneLib.buildDepsOnly craneArgs;
-        niri = craneLib.buildPackage (craneArgs // {inherit cargoArtifacts;});
+        niri = craneLib.buildPackage (craneArgs // {
+          inherit cargoArtifacts;
+          LD_LIBRARY_PATH = pkgs.lib.makeLibraryPath craneArgs.runtimeDependencies; # Needed for tests to find libxkbcommon
```

Comment:
Why is this being added here instead of to `craneArgs`, like with `LIBCLANG_PATH`?

Reply:
I didn't add it in `craneArgs` because `craneArgs` is also used for creating `cargoArtifacts`, and it's technically not necessary there.
I could also just move it to `craneArgs` though, it shouldn't make any difference, I simply tried to keep everything as clean as possible

Reply:
Also, if I put `LD_LIBRARY_PATH` into `craneArgs`, I'd have to turn `craneArgs` into a recursive attrset in order to access the `runtimeDependencies`

Reply:
>  I'd have to turn `craneArgs` into a recursive attrset in order to access the `runtimeDependencies`
 
 `craneArgs` is in scope though. you can just move the line up without changing a single char on it

Reply:
Huh, I didn't know that was possible. I rebased it now to use `craneArgs`, and also fixed the deprecation warning:

```
warning: `crane.lib.${system}` is deprecated. please use `(crane.mkLib nixpkgs.legacyPackages.${system})` instead
```